### ISSUE TITLE
Garbage value returned (Distributed / LowCardinality / Nullable)

### ISSUE THREAD

**Issue by champtar** [yandex/ClickHouse#3615]
Hi Clickhouse Team,

I'm running 86c59bf3c607098d0f7eb5e85d0cb3c59151d9b2 (was e3b0cecdec24d63f7abf5db5b847b33fdcf3452d) with a table like
```
CREATE TABLE analytics.vme1
(
    date Date,
    datetime DateTime,
...
    clientip FixedString(16),
...
    geoip_latitude LowCardinality(Nullable(Float64)) DEFAULT NULL,
    geoip_longitude LowCardinality(Nullable(Float64)) DEFAULT NULL
)
ENGINE = Distributed('analytics_cluster', '', 'vme1', rand())
```

When doing GROUP BY on Distributed LowCardinality Nullable I get garbage values
```
SELECT 
    geoip_latitude AS latitude, 
    geoip_longitude AS longitude, 
    uniq(clientip) AS uniqueip
FROM analytics.vme1 
GROUP BY 
    latitude, 
    longitude
ORDER BY uniqueip DESC
LIMIT 2

┌──────────────latitude─┬──────────────longitude─┬─uniqueip─┐
│                  ᴺᵁᴸᴸ │ 7.291122019556398e-304 │   153250 │
│ 3.105036184601418e231 │                   ᴺᵁᴸᴸ │    42062 │
└───────────────────────┴────────────────────────┴──────────┘

2 rows in set. Elapsed: 11.552 sec. Processed 1.45 billion rows, 29.80 GB (125.90 million rows/s., 2.58 GB/s.) 
```

If I request the shards directly everything work fine
```
SELECT 
    geoip_latitude AS latitude, 
    geoip_longitude AS longitude, 
    uniq(clientip) AS uniqueip
FROM analytics_shard_3.vme1 
GROUP BY 
    latitude, 
    longitude
ORDER BY uniqueip DESC
LIMIT 2

┌─latitude─┬─longitude─┬─uniqueip─┐
│     ᴺᵁᴸᴸ │      ᴺᵁᴸᴸ │   149148 │
│ -34.6033 │  -58.3816 │    41585 │
└──────────┴───────────┴──────────┘

2 rows in set. Elapsed: 7.712 sec. Processed 484.81 million rows, 9.94 GB (62.86 million rows/s., 1.29 GB/s.) 
```

Might be related to #3227 
I'll try to create a repro case soon

**Comment by champtar** [comment]
I have a "small" repro case, with 3 servers (actually 2 are enough I think)

create a cluster:
```
<yandex>
<remote_servers>
<test_cluster>
<shard>
    <weight>1</weight>
    <replica>
        <default_database>test</default_database>
        <host>server1</host>
        <port>9000</port>
        <user>distributed</user>
        <password>password</password>
    </replica>
</shard>
<shard>
    <weight>1</weight>
    <replica>
        <default_database>test</default_database>
        <host>server2</host>
        <port>9000</port>
        <user>distributed</user>
        <password>password</password>
    </replica>
</shard>
</test_cluster>
</remote_servers>
</yandex>
```

On each servers:
```
SET allow_experimental_low_cardinality_type = 1;
CREATE DATABASE test;
CREATE TABLE test.test_low_null_float ON CLUSTER test_cluster (a LowCardinality(Nullable(Float64))) ENGINE = Memory;
CREATE TABLE test.dist (a LowCardinality(Nullable(Float64))) ENGINE = Distributed('test_cluster', 'test', 'test_low_null_float', rand()) ;
```

Then insert enough data:
```
INSERT INTO test.dist (a) SELECT number FROM system.numbers LIMIT 10000000;
```

And query it
```
SELECT a, count() FROM test.dist GROUP BY a ORDER BY count() LIMIT 1;

┌───────────────────────a─┬─count()─┐
│ -1.557374211108995e-207 │   39061 │
└─────────────────────────┴─────────┘
1 rows in set. Elapsed: 2.192 sec. Processed 10.00 million rows, 120.00 MB (4.56 million rows/s., 54.73 MB/s.) 

```

**Comment by champtar** [comment]
Fixed with #3632

**Comment by champtar** [comment]
Thanks @KochetovNicolai
